kpconv_torch/datasets/S3DIS.py

```diff
@@ -865,20 +865,34 @@ def load_evaluation_points(self, file_path):
     def read_input(self, filename):
         """Read all the input files that belong to the dataset
 
-        Ply files are read by training and testing commands.
+        PLY files are read by training and testing commands.
         """
-        file_extension = filename.split(".")[-1]
-        if file_extension == "ply":
+        file_extension = Path(filename).suffix
+        if file_extension == ".ply":
             # Read ply file
             data = read_ply(filename)
             points = np.vstack((data["x"], data["y"], data["z"])).T
             colors = np.vstack((data["red"], data["green"], data["blue"])).T
-            labels = data["classification"]
-        elif file_extension == "xyz":
+            labels = data["class"]
+        elif file_extension == ".xyz":
             data = np.loadtxt(filename, delimiter=" ")
             points = data[:, :3].astype(np.float32)
             colors = data[:, 3:].astype(np.uint8)
             labels = np.zeros(shape=(data.shape[0],)).astype(np.int32)
+        elif file_extension == ".las" or file_extension == ".laz":
```

Comment:
Now we've introduced unit tests, it should be a really good idea to test this new feature! ;-)

Plus, you distinguish two different factors, it could be interesting to test both factor scenarios with accurate `.las` fixture files.

Reply:
what do you mean with "factors"?

Reply:
how would you generate las and laz fixture files ?

Reply:
> how would you generate las and laz fixture files ?

You can create them with the help of the `laspy` API. Or alternatively pick some `.las` on the internet (but we're talking about a unit test fixture, a small/controlable file is expected, then the hand-made `.las` is preferable).

Reply:
I have removed the factor, since the 1.3 las format forces the colors to be coded on two bytes, with a value * 256 relatively to the original one. 

Tests have been added, for las and xyz formats (in addition to the one already existing for the ply format).

Reply:
> I have removed the factor, since the 1.3 las format forces the colors to be coded on two bytes, with a value * 256 relatively to the original one.

We should be explicit in the documentation regarding the `.las` support, on this point.

Reply:
By the way, I've seen on the `.las` format specification that format 6/7 that you mention elsewhere in the code appears in version 1.4 of the format!